src/data-workspace/default-form.js

```diff
@@ -14,10 +15,16 @@ export function DefaultForm({ dataSet, globalFilterText }) {
     }
 
     const dataElements = selectors.getDataElementsByDataSetId(data, dataSet.id)
+    const indicators = selectors.getIndicatorsByDataSetId(data, dataSet.id)
     const groupedDataElements = selectors.getGroupedDataElementsByCatCombo(
         data,
         dataElements
     )
+    const nrColumnsInTable = groupedDataElements.reduce(
+        (max, { categoryCombo: { categoryOptionCombos } }) =>
```

Comment:
So we previously did this in `section.js` as well, but that can cause problems when the server returns the wrong `CategoryOptionCombos`. The table will then render empty padding-cells if the server returns more "cocs" than it should. 
Of course, this is a server-bug, but we fixed this problem in `section.js` by calculating the amount of `cocs` by the actual present `categoryOptions`:


```javascript
    // calculate how many columns in each group
    const groupedTotalColumns = groupedDataElements.map((grp) =>
        (
            selectors
                .getCategoriesByCategoryComboId(data, grp.categoryCombo.id)
                ?.map((cat) => cat.categoryOptions.length) || [1]
        ).reduce((total, curr) => total * curr)
    )

    const nrColumnsInTable = Math.max(...groupedTotalColumns)
```

Reply:
Thanks for pointing this out @Birkbjo 🙇 

I ended up implementing this functionality in its most simple form in https://github.com/dhis2/data-entry-app/pull/89/commits/d57812b123498c59224561cc37398169a454969f. 

I liked the idea of having a cached selector, but I couldn't get this to work. I tried just copy-pasting the last two snippets you posted but that didn't run. I got a bit 🤯 ... I think the complexity lies in the fact that this selector would also need the `groupedDataElements`, and these are computed conditionally...

Also, adding `useMemo` was problematic, because we're doing the early return for `isLoading || isError`, so that makes the hook conditional. Moving everything above the early return to avoid that also seemed a bit problematic. I decided to just leave things like this for now. This change wasn't completely out-of-scope anyway, I simply happened to find it when I was adding indicator support to default forms.

Reply:
> I liked the idea of having a cached selector, but I couldn't get this to work. I tried just copy-pasting the last two snippets you posted but that didn't run. 

Yeah, I'm sorry, I just wrote that code directly in the comment. Was missing a parenthesis at the end at the `keySelector`-function. Also missed a `y` in `cat.categoryOptions.length`... Will test my code next time!

> Also, adding useMemo was problematic, because we're doing the early return for isLoading || isError, so that makes the hook conditional. 

We should probably remove this regardless, as that is handled in `LoadApp`, and not necessary to handle further down the tree.